src/components/autocomplete-select/index.jsx

```diff
@@ -0,0 +1,232 @@
+  selectProps: PropTypes.object.isRequired,
+};
+
+function Option(props) {
+  const {
+    // Ignored for performance
+    // https://github.com/JedWatson/react-select/issues/3128
+    onMouseMove,
+    onMouseOver,
+
+    ...innerProps
+  } = props.innerProps;
+
+  return (
+    <MenuItem
+      ref={props.innerRef}
+      component="div"
+      style={{
+        fontWeight: props.isSelected ? 500 : 400,
+      }}
+      {...innerProps}
+    >
+      {props.children}
+    </MenuItem>
+  );
+}
+
+Option.propTypes = {
+  children: PropTypes.node,
+  innerProps: PropTypes.object,
+  innerRef: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
+  isFocused: PropTypes.bool,
+  isSelected: PropTypes.bool,
+};
+
+function Placeholder(props) {
+  return (
+    <Typography
+      color="textSecondary"
+      className={props.selectProps.classes.placeholder}
+      {...props.innerProps}
+    >
+      {props.children}
+    </Typography>
+  );
+}
+
+Placeholder.propTypes = {
+  children: PropTypes.node,
+  innerProps: PropTypes.object,
+  selectProps: PropTypes.object.isRequired,
+};
+
+function SingleValue(props) {
+  return (
+    <Typography className={props.selectProps.classes.singleValue} {...props.innerProps}>
+      {props.children}
+    </Typography>
+  );
+}
+
+SingleValue.propTypes = {
+  children: PropTypes.node,
+  innerProps: PropTypes.object,
+  selectProps: PropTypes.object.isRequired,
+};
+
+function ValueContainer(props) {
+  return <div className={props.selectProps.classes.valueContainer}>{props.children}</div>;
+}
+
+ValueContainer.propTypes = {
+  children: PropTypes.node,
+  selectProps: PropTypes.object.isRequired,
+};
+
+function Menu(props) {
+  return (
+    <Paper square className={props.selectProps.classes.paper} {...props.innerProps}>
+      {props.children}
+    </Paper>
+  );
+}
+
+Menu.propTypes = {
+  children: PropTypes.node,
+  innerProps: PropTypes.object,
+  selectProps: PropTypes.object,
+};
+
+const components = {
+  Control,
+  Menu,
+  NoOptionsMessage,
+  Option,
+  Placeholder,
+  SingleValue,
+  ValueContainer,
+};
+
+class AutocompleteSelect extends Component {
+  static propTypes = {
+    options: PropTypes.arrayOf(PropTypes.string).isRequired,
+    value: PropTypes.string,
+    onChange: PropTypes.func.isRequired,
+    name: PropTypes.string.isRequired,
+    helperText: PropTypes.string,
+  };
+
+  render() {
+    const {classes, options, value, onChange, name, helperText, ...props} = this.props;
+
+    return (
+      <Select
+        classes={classes}
+        components={components}
+        options={options.map(option => ({value: option, label: option}))}
+        value={value && {value, label: value}}
+        onChange={({value}) => onChange({target: {name, value}})}
+        TextFieldProps={{
+          helperText,
+          margin: 'normal',
+          variant: 'filled',
+        }}
+        {...props}
+      />
+    );
+  }
+}
+
+export default withStyles(styles)(AutocompleteSelect);
```

Comment:
Personally I think this is hard to read there's a ton going on in this file. But also don't think it's worth delaying the feature to change.

Reply:
Totally agreed - it's mostly a direct lift from [material-ui's react-select implementation](https://material-ui.com/components/autocomplete/#react-select) and I figured it wasn't worth changing much for a single-use component